python/bind_slice_methods.h

```diff
@@ -102,16 +104,19 @@ template <class T> struct slicer {
         }
         auto start = py::getattr(py_slice, "start");
         auto stop = py::getattr(py_slice, "stop");
-        auto start_var =
-            start.is_none()
-                ? VariableConstView{}
-                : py::getattr(py_slice, "start").cast<VariableConstView>();
-        auto stop_var =
-            stop.is_none()
-                ? VariableConstView{}
-                : py::getattr(py_slice, "stop").cast<VariableConstView>();
-
-        return slice(self, dim, start_var, stop_var);
+        if (!start.is_none() || !stop.is_none()) { // Means default slice : is
```

Comment:
I don't think this check is strictly needed as if you pass two empty VariableConstView objects to the slice implementation it should default through to the index based slicing anyway. You are exiting early and avoiding some unnecessary code but at a slight complexity cost here. Not sure which is better, what was your reasoning?

Reply:
Prior to these changes if you did this
```python
ds['x', :]
```
It would interpret as value based slicing not index based slicing since `:` is equivalent to `slice(None, None, None)` and therefore an empty `VariableConstView` is made for start and stop and it is treated as value based slicing.

Reply:
Note https://dev.azure.com/scipp/scipp/_build/results?buildId=1441&view=logs&jobId=f2a3abc3-f415-54ca-77f6-7b36a30e229e&j=f2a3abc3-f415-54ca-77f6-7b36a30e229e&t=9dfee2f3-c920-5294-e9b7-bf7c2ad77947 alerted me to this problem

Reply:
Yes had missed the commit message specifying that there are some cases where defaulting through value based slicing is more restrictive. Now makes sense.